app/components/chat_component.rb

```diff
@@ -0,0 +1,45 @@
+class ChatComponent < ViewComponent::Base
+  include Motion::Component
+
+  attr_reader :user, :conversation, :messages, :send_message_id
+
+  def initialize(user:, conversation:, messages:)
+    @user = user
+    @conversation = conversation
+    @messages = messages
+    @send_message_id = SecureRandom.uuid
+    self.class.stream_for(conversation, :handle_received)
```

Comment:
this looks weird, but I couldn't do it outside the `initialize` method since I needed the `conversation`

Reply:
`stream_for` is both a class and instance method for this reason. The class method is just a shorthand for calling it on all instances.

I think you should be able to do `stream_for(conversation, :handle_received)` here (`self.class` will cause all future instances to listen to that stream too!).